templates/auth_gssapi.conf.erb

```diff
@@ -13,3 +13,16 @@
   ErrorDocument 500 '<html><meta http-equiv="refresh" content="0; URL=/users/login"><body>Kerberos authentication did not pass.</body></html>'
 </Location>
 
+<Location /api/users/extlogin>
+  SSLRequireSSL
+  AuthType GSSAPI
+  AuthName "GSSAPI Single Sign On Login"
+  GssapiCredStore keytab:<%= @http_keytab %>
+  GssapiSSLonly On
+  GssapiLocalName On
+  # require valid-user
+  require pam-account <%= scope.lookupvar('foreman::pam_service') %>
+  ErrorDocument 401 '<html><meta http-equiv="refresh" content="0; URL=/users/login"><body>Kerberos authentication did not pass.</body></html>'
+  # The following is needed as a workaround for https://bugzilla.redhat.com/show_bug.cgi?id=1020087
+  ErrorDocument 500 '<html><meta http-equiv="refresh" content="0; URL=/users/login"><body>Kerberos authentication did not pass.</body></html>'
```

Comment:
Since this is an API endpoint, maybe not returning html on errrors would be a good idea?

Reply:
I thought so as well, but after changes per Ewoud suggestions it's not obvious to me how to distinguish between UI vs API...